user/store.go

```diff
@@ -104,9 +106,11 @@ func NewStore(ctx context.Context, db *sql.DB) (*Store, error) {
 
 		findOne: p.P(`
 			SELECT
-				id, name, email, avatar_url, role, alert_status_log_contact_method_id
-			FROM users
-			WHERE id = $1
+				u.id, u.name, u.email, u.avatar_url, u.role, u.alert_status_log_contact_method_id, fav is distinct from null
+			FROM users u
+			LEFT JOIN user_favorites fav ON
+				fav.tgt_user_id = u.id AND fav.user_id = $2
+			WHERE u.id = $1
 		`),
 		findOneForUpdate: p.P(`
```

Comment:
Since `findOneForUpdate` doesn't return the user favorite status, the row returned from the query should be scanned into a user object directly with `row.Scan()` instead of `u.scanFrom()`.

Reply:
Addressed, fixed bug by implementing `row.Scan()` if `forUpdate` returns true. Previous behavior observed prompted an error when editing a single user role to Admin.